packetman: fix "TCP-dataoffset" tranformation and "TCP-checksum" tranformation

Lanius-collaris opened the pull request.

Lanius-collaris committed:
packetman: fix docs of PacketSpecs

Lanius-collaris committed:
Create contributors/Lanius-collaris.md

Lanius-collaris committed:
packetman: fix "TCP-dataoffset" tranformation and "TCP-checksum" tranformation

Lanius-collaris commented:
@rod-hynes

rod-hynes commented:
Thanks for contributing this fix. I think your new logic looks correct.

At this point, I'm not sure the existing `transformation.setsCalculatedField` or new `transformation.isDataOffset`/`transformation.isChecksum` abstractions are adding much value, and we could simplify the code by dropping all of those functions from `transformation` and instead just check the special cases in `apply`; roughly something like:

```
if t, ok := transform.(*transformationTCPField); ok && t.fieldName == tcpFieldDataOffset {
	fixLengths = false
	setCalculatedField = true
}
if t, ok := transform.(*transformationTCPField); ok && t.fieldName == tcpFieldChecksum {
	computeChecksums = false
	setCalculatedField = true
}
```

Does that make sense?

Lanius-collaris committed:
packetman: simplify type transformation

drop transformation.setsCalculatedField(), transformation.isDataOffset(), transformation.isChecksum()
